untie/src/untie.cpp

```diff
@@ -25,26 +25,46 @@ Untie::Untie() {
     }
 
     {
-        auto cubeObj = scene.create_game_object("cube_0");
+        auto cubeObj = scene.create_game_object("ground");
         cubeObj.get_component<components::Transform>().set_position(glm::vec3(0, -10, 0));
         cubeObj.get_component<components::Transform>().set_scale(glm::vec3(15.0f, 1.0f, 15.0f));
         auto& mesh = cubeObj.add_component<components::Mesh>();
         mesh.create_cube();
         auto material = cubeObj.add_component<components::Material>();
+        auto& rigidbody = cubeObj.add_component<components::RigidBody>();
+        rigidbody.set_physics_and_scene(m_engine->get_physics_moddule()->get_active_Scene(),
+                                        m_engine->get_physics_moddule()->get_physics());
+        rigidbody.set_transform(cubeObj.get_component<components::Transform>().get_position(),
+                                cubeObj.get_component<components::Transform>().get_rotation());
```

Comment:
Can you move this to the `on_awake` function? You can also detect if the same `GameObject` contains the `Transform` component and automatically set it from there.

Reply:
The `physics` and `scene` are got from the `Engine`. I have not found a way to directly get the `physics` module from the `Engine`. if I move `set_physics_and_scene`  to `on_awake`, the `RigidBody` component needs to save the engine as variables.

And for `Transform`. i tried code `get_position_from_transform` in `RigidBody` component. But it shows **unresolved external symbol** in `get_game_object_from_component`